l1-contracts/contracts/chain-registrar/ChainRegistrar.sol

```diff
@@ -0,0 +1,178 @@
+    ) public {
+        ChainConfig memory config = ChainConfig({
+            chainId: _chainId,
+            pubdataPricingMode: _pubdataPricingMode,
+            blobOperator: _blobOperator,
+            operator: _operator,
+            governor: _governor,
+            baseToken: BaseToken({
+                tokenAddress: _tokenAddress,
+                tokenMultiplierSetter: _tokenMultiplierSetter,
+                gasPriceMultiplierNominator: _gasPriceMultiplierNominator,
+                gasPriceMultiplierDenominator: _gasPriceMultiplierDenominator
+            })
+        });
+        if (
+            deployedChains[msg.sender][config.chainId] || bridgehub.stateTransitionManager(config.chainId) != address(0)
+        ) {
+            revert ChainIsAlreadyDeployed();
+        }
+        proposedChains[msg.sender][_chainId] = config;
+        // For Deploying L2 contracts on for non ETH based networks, we as bridgehub owners required base token.
+        if (config.baseToken.tokenAddress != ETH_TOKEN_ADDRESS) {
+            uint256 amount = (1 ether * config.baseToken.gasPriceMultiplierNominator) /
+                config.baseToken.gasPriceMultiplierDenominator;
+            if (IERC20(config.baseToken.tokenAddress).balanceOf(l2Deployer) < amount) {
+                IERC20(config.baseToken.tokenAddress).safeTransferFrom(msg.sender, l2Deployer, amount);
+            }
+        }
+        emit NewChainRegistrationProposal(config.chainId, msg.sender);
+    }
+
+    // @dev Change l2 deployer
+    function changeDeployer(address _newDeployer) public onlyOwner {
+        l2Deployer = _newDeployer;
+        emit L2DeployerChanged(l2Deployer);
+    }
+
+    function getChainConfig(address _author, uint256 _chainId) public view returns (ChainConfig memory) {
+        return proposedChains[_author][_chainId];
+    }
+
+    // @dev Get data about the chain that has been fully deployed
+    function getRegisteredChainConfig(uint256 _chainId) public view returns (RegisteredChainConfig memory) {
+        address stm = bridgehub.stateTransitionManager(_chainId);
+        if (stm == address(0)) {
+            revert ChainIsNotYetDeployed();
+        }
+        address diamondProxy = IStateTransitionManager(stm).getHyperchain(_chainId);
+
+        address pendingChainAdmin = IGetters(diamondProxy).getPendingAdmin();
+        address chainAdmin = IGetters(diamondProxy).getAdmin();
+        address l2BridgeAddress = bridgehub.sharedBridge().l2BridgeAddress(_chainId);
+        if (l2BridgeAddress == address(0)) {
+            revert BridgeIsNotRegistered();
+        }
+
+        RegisteredChainConfig memory config = RegisteredChainConfig({
+            pendingChainAdmin: pendingChainAdmin,
+            chainAdmin: chainAdmin,
+            diamondProxy: diamondProxy,
+            l2BridgeAddress: l2BridgeAddress
+        });
+        return config;
+    }
+
+    // @dev Mark chain as registered. Emit necessary events for spinning up the chain server
```

Comment:
I still don't understand why would multisig call this function that... emit events and thats it. All events are emitted on actual deployment so no need to do so.